examples/all-clusters-app/esp32/CMakeLists.txt

```diff
@@ -20,13 +20,22 @@ cmake_minimum_required(VERSION 3.5)
 include($ENV{IDF_PATH}/tools/cmake/project.cmake)
 
 # The list of extra component dirs must be in sync with that in all-clusters-app/esp32/Makefile
+if(${IDF_TARGET} STREQUAL "esp32")
 set(EXTRA_COMPONENT_DIRS
     "${CMAKE_CURRENT_LIST_DIR}/third_party/connectedhomeip/config/esp32/components"
     "${CMAKE_CURRENT_LIST_DIR}/../../common/m5stack-tft/repo/components"
     "${CMAKE_CURRENT_LIST_DIR}/../../common/QRCode"
     "${CMAKE_CURRENT_LIST_DIR}/../../common/screen-framework"
 )
+elseif(${IDF_TARGET} STREQUAL "esp32c3")
+set(EXTRA_COMPONENT_DIRS
```

Comment:
Whether the screen display components should be added shouldn't depend on the chip model (say if the m5stack has a esp32c3 version). Maybe we shall use `DEVICE_TYPE_` Kconfig options instead and disable these components for  ESP32-DevKitC and ESP32-DevKitM?

Reply:
the screen display components will be compile when they are added to `EXTRA_COMPONENT_DIRS`, and the `DEVICE_TYPE_` Kconfig options can't be used by the top level CMakeLists.txt.